cmd/source_profile.go

```diff
@@ -0,0 +1,274 @@
+package cmd
+
+import (
+	"fmt"
+	"os"
+	"strconv"
+	"strings"
+)
+
+type SourceProfileType int
+
+const (
+	SourceProfileTypeUnset = iota
+	SourceProfileTypeFile
+	SourceProfileTypeConnection
+	SourceProfileTypeConfig
+)
+
+type SourceProfileFile struct {
+	path   string
+	format string
+}
+
+func NewSourceProfileFile(params map[string]string) SourceProfileFile {
+	profile := SourceProfileFile{}
+	if !filePipedToStdin() {
+		profile.path = params["file"]
+	}
+	if format, ok := params["format"]; ok {
+		profile.format = format
+		// TODO: Add check that format takes values from ["dump", "csv", "avro", ... etc]
+	} else {
+		fmt.Printf("User did not specify `format` in source-profile, defaulting to \"dump\"\n")
```

Comment:
Suggest rephrasing this as:

"source-profile format defaulting to \"dump\"\n"

Reply:
Done.